de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/editor/StoppableDocumentListener.java

```diff
@@ -1,16 +1,18 @@
 package de.fu_berlin.inf.dpp.intellij.editor;
 
+import com.intellij.openapi.editor.Document;
 import com.intellij.openapi.editor.EditorFactory;
 import com.intellij.openapi.editor.event.DocumentEvent;
 import com.intellij.openapi.editor.event.DocumentListener;
+import com.intellij.openapi.fileEditor.FileDocumentManager;
+import com.intellij.openapi.vfs.VirtualFile;
 import de.fu_berlin.inf.dpp.activities.SPath;
 import org.apache.log4j.Logger;
 
 /**
- * Tracks modifications of documents opened in editors.
- * <p/>
- * Notify its EditorManager of modifications in an opened document, before
- * the modification has actually happened.
+ * Tracks modifications of Documents and triggers matching TextEditActivities.
```

Comment:
Just FYI

In Eclipse

1. IIRC we act on after change
2. Refactoring changes are mostly broadcasted through file activities if there is not open editor / connected document provider for a given file.

Reply:
Regarding point 1: I just left it the way it was. Is it important to do it after the change? This could be changed easily, but I don't know if this could cause any issues.

As a side note: The new filesystem handling also acts before files are deleted/moved/renamed. Is it important for this part of the logic (also thinking about the future goal of cross-IDE compatibility)?

Regarding point 2: This is not the case in IntelliJ. I added this logic as text edits changing the package defined in the java were not send as there was no editor open for the file at the point they were created. I would guess that IntelliJ closes the old editor, moves the file, adjusts the content through a document for the new file and then re-opens the moved file in an editor.